string robot_name
string source
string dest